File: social-app-backend/src/main/java/razepl/dev/socialappbackend/auth/apicalls/RegisterRequest.java
package razepl.dev.socialappbackend.auth.apicalls;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import razepl.dev.socialappbackend.auth.interfaces.RegisterUserRequest;

import java.time.LocalDate;

/**
 * The RegisterRequest class is a model class representing a user registration request.
 * This class includes user information and implements the {@link RegisterUserRequest} interface.
 */
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RegisterRequest implements RegisterUserRequest {
    private String name;
    private String surname;
    private String email;
    private String password;
    private LocalDate dateOfBirth;
}

File: social-app-frontend/src/app/core/data/register-request.ts
/**
 * Represents a registration request to be sent to the server.
 */
export class RegisterRequest {
    /**
     * The user's first name.
     */
    name!: string;

    /**
     * The user's last name.
     */
    surname!: string;

    /**
     * The user's email address.
     */
    email!: string;

    /**
     * The user's chosen password.
     */
    password!: string;

    /**
     * The user's date of birth.
     */
    dateOfBirth!: Date;
}
